lib/control/control-commands.c

```diff
@@ -58,6 +58,26 @@ control_register_command(const gchar *command_name, const gchar *description, Co
   command_list = g_list_append(command_list, new_command);
 }
 
+void
+control_replace_command(const gchar *command_name, const gchar *description, CommandFunction function,
+                        gpointer user_data)
+{
+  GList *command_it =  g_list_find_custom(command_list, command_name,
+                                          (GCompareFunc)control_command_start_with_command);
+  if (!command_it)
+    {
+      msg_warning("Trying to replace a non-existent command. Command will be registered as a new command.",
+                  evt_tag_str("command", command_name));
+      control_register_command(command_name, description, function, user_data);
```

Comment:
Do we need the warning here? 

If we do, I think we need a bit more context, like mentioning that this was a control socket command, but I like the idea that modules can register such stuff.

But all in all I would make this msg_debug().

Reply:
changed to debug